src/moscot/problems/cross_modality/_mixins.py

```diff
@@ -0,0 +1,181 @@
+        ...
+
+
+class CrossModalityTranslationMixin(AnalysisMixin[K, B]):
+    """Cross modality translation analysis mixin class."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self._src_attr: Optional[Dict[str, Any]] = None
+        self._tgt_attr: Optional[Dict[str, Any]] = None
+
+    def translate(  # type: ignore[misc]
+        self: CrossModalityTranslationMixinProtocol[K, B],
+        source: K,
+        target: K,
+        forward: bool = True,
+        alternative_attr: Optional[Dict[str, Any]] = None,
+        **kwargs: Any,
+    ) -> ArrayLike:
+        """Translate source modality to target modality.
+
+        Parameters
+        ----------
+        source
+            Key identifying the source distribution.
+        target
+            Key identifying the target distribution.
+        forward
+            If `True`, compute the translation from :attr:`adata_src` to :attr:`adata_tgt`, otherwise vice-versa.
+        alternative_attr
+            Optional parameter to specify an alternative embedding to translate. If ``forward = True` and`
+            - if :class:`str`, it must refer to a key in :attr:`~anndata.AnnData.obsm` in the source distribution.
+            - if :class:`dict`, the dictionary stores `attr` (attribute of :class:`~anndata.AnnData`) and `key`
+              (key of :class:`AnnData.{attr} <anndata.AnnData>`) in the source distribution.
+        kwargs
+            Keyword arguments for policy-specific `_apply` method of :class:`~moscot.base.problems.CompoundProblem`.
+
+        Returns
+        -------
+        Translation from :attr:`adata_src` in target domain or from :attr:`adata_tgt` in source domain,
+        depending on `forward`.
+        """
+
+        def _get_features(
+            adata: AnnData,
+            attr: Dict[str, Any],
+        ) -> ArrayLike:
+            data = getattr(adata, attr["attr"])
+            key = attr.get("key", None)
+            return data if key is None else data[key]
+
+        if self._src_attr is None:
+            raise ValueError("Source attribute is None.")
+        if self._tgt_attr is None:
+            raise ValueError("Target attribute is None.")
+
+        kwargs["scale_by_marginals"] = True
+        kwargs["normalize"] = False
+
+        prob = self[source, target]  # type: ignore[index]
+        if alternative_attr is None:
+            src_attr = self._src_attr
+            tgt_attr = self._tgt_attr
+        else:
+            src_attr = tgt_attr = (
+                {"attr": "obsm", "key": alternative_attr} if isinstance(alternative_attr, str) else alternative_attr
+            )
+
+        if not forward:
+            return prob.pull(_get_features(self.adata_tgt, attr=tgt_attr), **kwargs)
+
+        adata_src = self.adata_src if self.batch_key is None else prob.adata_src
+        return prob.push(_get_features(adata_src, attr=src_attr), **kwargs)
```

Comment:
@michalk8 @MUCDK 
You switched `pull` and `push` now, right? Still wondering, why it wasn't correct the way it was before..? 

As I mentioned in the last comment on this, accoring to the docs, the `push` function computes `x_new = T^T x ` where T is the transport matrix, and T^T its transpose.

The forward `translate` function yields the barycentric projection, that maps src to tgt embedding, i.e. it should compute `x_new = T x` (ignoring scaling). However, this calculation corresponds to moscots `pull` function, that's why I used `pull` in the forward calculation.

Also refer to [SCOT Paper](https://www.biorxiv.org/content/10.1101/2020.04.28.066787v2.full) for the definition of the barycentric projection that I want to compute here.

Reply:
@nosander thanks, we just discussed this, and you're right.